packages/api/src/middleware/validation/checkAnimalOrBatch.js

```diff
@@ -572,9 +609,18 @@ export function checkRemoveAnimalOrBatch(animalOrBatchKey) {
         if (!preexistingAnimalOrBatch) {
           invalidIds.push(animalOrBatch.id);
         }
+        removalDatesSet.add(animalOrBatch.removal_date);
       }
 
       await checkInvalidIds(invalidIds);
+
+      // Assumption: All removal_date values are identical.
+      // This check ensures that if this assumption ever changes, it triggers an error.
+      // If the error is triggered, re-implement handleIncompleteTasksForAnimalsAndBatches to handle multiple dates.
+      if (removalDatesSet.size > 1) {
+        throw customError('removal_date is expected to be the same in all animals/batches');
+      }
```

Comment:
This might have been a bad decision, but I assumed each animal/batch wouldn't have a different `removal_date`.